src/brod_producer.erl

```diff
@@ -221,11 +232,23 @@ handle_info({msg, Pid, CorrId, #produce_response{} = R},
         "Offset: ~B\n"
         "Error code: ~p",
         [Topic, Partition, Offset, ErrorCode]),
-      {stop, {produce_error, {Topic, Partition, Offset, ErrorCode}}, State};
+      Error = {produce_response_error, Topic, Partition, Offset, ErrorCode},
+      {ok, NewState} = schedule_retry(State, Error),
```

Comment:
since we have a -1 retry times.
don't we want to be selective on errors to retry ?